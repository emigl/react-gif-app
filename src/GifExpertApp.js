import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'
// Se ha tenido que agregar un default prop para poder realizar las pruebas, así es mucho más sencillo manejar los componentes también
const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories)

    // const handleAdd = () => {
    //     // Para usar setCategories se necesita utilizar el operador spread del array que NO hay que mutar, y con ese operador crea una copia de ese array u objeto, según lo que se haya creado, se necesitan las llaves [] para decirle que es un array, si categories fuera un objeto supongo que habría que utilizar {}
    //     setCategories([...categories,'HunterXHunter'])

    //     // Al recibir un callback la funcion, el parametro "cats" sería la variable "categories" en este caso, es decir, que el callback llama a la funcion donde se almacena la información del hook useState
    //    setCategories( cats => [...cats, 'HunterXHunter'])
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            
                {
                    categories.map( category => (
                        <GifGrid 
                        key = {category}
                        category = {category} />
                    ))
                }
            
        </>
    )
}

export default GifExpertApp
