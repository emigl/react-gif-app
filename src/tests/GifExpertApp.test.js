import {shallow} from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Test de GifExpertApp component', () => {
    


    test('debe de hacer match con el snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot()
    })
    
    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    })
    
})
