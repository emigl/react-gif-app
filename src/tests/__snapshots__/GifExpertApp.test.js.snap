// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifExpertApp component debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Test de GifExpertApp component debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;
